src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,368 @@
+                  </div>
+                </div>
+              </div>
+            </Reference>
+          </div>
         </div>
       </div>
     </div>
-  </div>
+  </GenericModal>
 </template>
 
 <script>
-import debounce from 'docc-render/utils/debounce';
+import NavigatorLeafIcon from 'docc-render/components/Navigator/NavigatorLeafIcon.vue';
+import FilterInput from 'docc-render/components/Filter/FilterInput.vue';
+import GenericModal from 'docc-render/components/GenericModal.vue';
 import QuickNavigationHighlighter from 'docc-render/components/Navigator/QuickNavigationHighlighter.vue';
+import MagnifierIcon from 'theme/components/Icons/MagnifierIcon.vue';
 import Reference from 'docc-render/components/ContentNode/Reference.vue';
+import debounce from 'docc-render/utils/debounce';
+import keyboardNavigation from 'docc-render/mixins/keyboardNavigation';
+import symbolTreeNavigator from 'docc-render/mixins/symbolTreeNavigator';
 
 export default {
   name: 'QuickNavigationModal',
   components: {
+    FilterInput,
+    GenericModal,
+    MagnifierIcon,
+    NavigatorLeafIcon,
     QuickNavigationHighlighter,
     Reference,
   },
-  inject: ['quickNavigationStore'],
+  mixins: [
+    keyboardNavigation,
+    symbolTreeNavigator,
+  ],
   data() {
     return {
       debouncedInput: '',
-      quickNavigationStore: this.quickNavigationStore,
       userInput: '',
     };
   },
+  props: {
+    children: {
+      type: Array,
+      required: true,
+    },
+    showQuickNavigationModal: {
+      type: Boolean,
+      required: true,
+    },
+  },
   computed: {
     filteredSymbols: ({
-      flattenIndex,
       constructFuzzyRegex,
+      children,
       fuzzyMatch,
       debouncedInput,
       orderSymbolsByPriority,
     }) => {
-      const symbols = flattenIndex.filter(symbol => (
+      const symbols = children.filter(symbol => (
         symbol.type !== 'groupMarker'
         && symbol.title != null
       ));
       if (!debouncedInput) return [];
       const matches = fuzzyMatch({
-        debouncedInput: debouncedInput.toLowerCase(),
+        debouncedInput,
         symbols,
-        processedInputRegex: new RegExp(constructFuzzyRegex(debouncedInput)),
+        processedInputRegex: new RegExp(constructFuzzyRegex(debouncedInput), 'i'),
       });
       // Return the first 20 symbols out of sorted ones
       return orderSymbolsByPriority(matches).slice(0, 20);
     },
-    flattenIndex: ({ quickNavigationStore }) => quickNavigationStore.state.flattenIndex,
+    noResultsWereFound: ({ debouncedInput, totalItemsToNavigate }) => (
+      debouncedInput.length && !totalItemsToNavigate
+    ),
+    totalItemsToNavigate: ({ filteredSymbols }) => filteredSymbols.length,
+  },
+  watch: {
+    userInput: 'debounceInput',
+    focusedIndex: 'scrollIntoView',
   },
   methods: {
     closeQuickNavigationModal() {
-      this.quickNavigationStore.toggleShowQuickNavigationModal();
+      this.$emit('closeQuickNavigationModal', false);
```

Comment:
Maybe just leverage the `:showQuickNavigationModal.sync` syntax?

```suggestion
      this.$emit('update:showQuickNavigationModal', false);
```

Reply:
Resolved 4bd2fd13e893ecb2b3dee9617a6ba7e430882f48